relayer/chain.go

```diff
@@ -222,12 +223,18 @@ func (src *Chain) BuildAndSignTx(datagram []sdk.Msg) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-
-	return auth.NewTxBuilder(
+	ctx := sdkCtx.CLIContext{Client: src.Client}
+	txBldr := auth.NewTxBuilder(
 		auth.DefaultTxEncoder(src.Amino.Codec), acc.GetAccountNumber(),
-		acc.GetSequence(), src.Gas, src.GasAdjustment, false, src.ChainID,
-		src.Memo, sdk.NewCoins(), src.getGasPrices()).WithKeybase(src.Keybase).
-		BuildAndSign(src.Key, ckeys.DefaultKeyPass, datagram)
+		acc.GetSequence(), src.Gas, src.GasAdjustment, true, src.ChainID,
+		src.Memo, sdk.NewCoins(), src.getGasPrices()).WithKeybase(src.Keybase)
+	if src.GasAdjustment > 0 {
```

Comment:
Why the `> 0` check? I'd assume if GasAdjustment isn't set (set to 0) then run the simulation otherwise use the specified value

Reply:
To allow backwards compatibility most examples and users of the `rly` already have configs in place without the `gas-adjustment` so it will be zero, EnrichWithGas at the end multiplies `simulation * gasAdjusment`